utils/src/hex_utils.ts

```diff
@@ -75,7 +75,7 @@ function slice(n: Numberish, start: number, end?: number): string {
 /**
  * Get the keccak hash of some data.
  */
-function hash(n: Numberish): string {
+function hash(n: Numberish | Buffer): string {
     return ethUtil.bufferToHex(ethUtil.keccak256(ethUtil.toBuffer(hexUtils.toHex(n))));
```

Comment:
Could we do something like...?
```suggestion
    const buf = Buffer.isBuffer(n) ? n : ethUtil.toBuffer(hexUtils.toHex(n));
    return ethUtil.bufferToHex(ethUtil.keccak256(buf));
```